### ISSUE TITLE
Whisper in web-llm with WebGPU?

### ISSUE THREAD

**Issue by sandorkonya** [mlc-ai/web-llm#68]
Great Repository!

Is it within your scope to implement a webGPU accelerated version of Whisper?

Not sure if this helps, but there is a [C port for Whisper wirh CPU implementation](https://github.com/ggerganov/whisper.cpp), and as mentioned in [this discussion](https://github.com/ggerganov/whisper.cpp/discussions/126), the main thing that needs to be offloaded to the GPU is the GGML_OP_MUL_MAT operator.

thy

**Comment by sandorkonya** [comment]
@tqchen my ultimate goal would be to get it run the most efficient way on android edge device. 

Although there is already a solution in the onnx framework [onnx framework](https://github.com/xenova/transformers.js/issues/100#issuecomment-1521661452), based on the [recent merge](https://github.com/microsoft/onnxruntime/pull/14579), but i am not sure when it will be usable on android.

There were some who tried with [GPU delegates](https://github.com/usefulsensors/openai-whisper/issues/36#issuecomment-1521977972), but no success yet.

Any idea how one could solve it on the edge (Android) device?

**Comment by DustinBrett** [comment]
There is also a demo of Whisper running via WebAssembly in that repo. https://github.com/ggerganov/whisper.cpp/tree/master/examples/talk.wasm

**Comment by sandorkonya** [comment]
> There is also a demo of Whisper running via WebAssembly in that repo. https://github.com/ggerganov/whisper.cpp/tree/master/examples/talk.wasm

Yes, it runs on CPU. I hope, that with a GPU version one could reach real time inference.
